appinventor/blocklyeditor/src/workspace_svg.js

```diff
@@ -1012,6 +1012,38 @@ Blockly.WorkspaceSvg.prototype.customContextMenu = function(menuOptions) {
   };
   menuOptions.push(enableAll);
 
+  // Clear unused blocks
+  var clearUnusedBlocks = {enabled: true};
+  clearUnusedBlocks.text = Blockly.Msg.REMOVE_UNUSED_BLOCKS;
+  clearUnusedBlocks.callback = function() {
+    Blockly.Events.setGroup(true);
+    var allBlocks = Blockly.getMainWorkspace().getTopBlocks()
+    var removeList = []
+    for (var x = 0, block; block = allBlocks[x]; x++) {
+      if(block.previousConnection != null || (block.outputConnection && block.outputConnection != null)) {
```

Comment:
It seems like `if (block.previousConnection || block.outputConnection)` would be a sufficient check here because these ought to only be set for blocks that have one of those (which the valid top-level blocks do not).